sdks/go/pkg/beam/util/gcsx/gcs.go

```diff
@@ -63,6 +64,41 @@ func Upload(ctx context.Context, client *storage.Client, project, bucket, object
 
 }
 
+// SoftDeletePolicyEnabled checks whether soft delete policy is enabled.
+// Note that soft delete policy is disabled when RetentionDuration is zero. Otherwise,
+// soft delete policy is enabled by default.
+func SoftDeletePolicyEnabled(attrs *storage.BucketAttrs) bool {
```

Comment:
Consider whether this function (and the other one being added) needs to be exported. Are we going to call this outside of this package? Are users ever expected to call it themselves?

If there's no need for it to be on the public user surface, then it shouldn't be exported. (Package level names with an initial Capital letter indicates exported or not, in Go).

Note you can always call unexported functions within the test files for the same package.